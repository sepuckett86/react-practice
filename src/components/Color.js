import React from 'react';

export default function Color() {
  const color = {
    name: 'Purple',
    hex: '#7b42f5',
    rgb: {
      red: 123,
      green: 66,
      blue: 245
    }
  };

  // ems for font
  // rem for padding, margin, width, etc.

  // const details = Object.keys(color)
  //   .map(detail => {
  //     return (
  //     <>
  //       <dt>{detail}</dt>
  //       <dd>{color[detail]}</dd>
  //     </>
  //     );
  //   });

  const style = {
    backgroundColor: color.hex,
    height: '0.8rem',
    width: '0.8rem',
    display: 'inline-block'
  };

  return (
    <dl>
      <dt>Name:</dt>
      <dd>{color.name}{' '}
        <div style={style}></div>
      </dd>
      <dt>Hex:</dt>
      <dd>{color.hex}</dd>

      <dt>RGB:</dt>
      <dd>
        <p>Red: {color.rgb.red}</p>
        <p>Green: {color.rgb.green}</p>
        <p>Blue: {color.rgb.blue}</p>
      </dd>
    </dl>
  );
}

