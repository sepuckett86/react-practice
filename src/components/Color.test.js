import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
