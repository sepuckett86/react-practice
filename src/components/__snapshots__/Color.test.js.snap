// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders App 1`] = `
<dl>
  <dt>
    Name:
  </dt>
  <dd>
    Purple
     
    <div
      style={
        Object {
          "backgroundColor": "#7b42f5",
          "display": "inline-block",
          "height": "0.8rem",
          "width": "0.8rem",
        }
      }
    />
  </dd>
  <dt>
    Hex:
  </dt>
  <dd>
    #7b42f5
  </dd>
  <dt>
    RGB:
  </dt>
  <dd>
    <p>
      Red: 
      123
    </p>
    <p>
      Green: 
      66
    </p>
    <p>
      Blue: 
      245
    </p>
  </dd>
</dl>
`;
